src/validation.cpp

```diff
@@ -4984,3 +5029,33 @@ void ChainstateManager::MaybeRebalanceCaches()
         }
     }
 }
+
+CBlockIndex* ChainstateManager::getSnapshotBaseBlock()
+{
+    auto blockhash_op = SnapshotBlockhash();
+    if (!blockhash_op) return nullptr;
+    return m_blockman.LookupBlockIndex(*blockhash_op);
+}
+
+std::optional<int> ChainstateManager::getSnapshotHeight()
+{
+    CBlockIndex* base = getSnapshotBaseBlock();
+    return base ? std::make_optional(base->nHeight) : std::nullopt;
+}
+
+CChainState& ChainstateManager::getChainstateForIndexing()
+{
+    return getSnapshotBaseBlock() ? *m_ibd_chainstate : *m_active_chainstate;
+}
+
+bool ChainstateManager::hasBgChainstateInUse()
+{
+    return this->SnapshotBlockhash() && !m_snapshot_validated;
+}
+
+bool ChainstateManager::IsAnyChainInIBD()
+{
+    return
```

Comment:
0ef00576cc9a4aac27310b9fc29c534ad829aad9 

```diff
 CChainState& ChainstateManager::getChainstateForIndexing()
 {
+    AssertLockHeld(::cs_main);
     return getSnapshotBaseBlock() ? *m_ibd_chainstate : *m_active_chainstate;
 }
 
@@ -5117,6 +5119,7 @@ bool ChainstateManager::hasBgChainstateInUse()
 
 bool ChainstateManager::IsAnyChainInIBD()
 {
+    AssertLockHeld(::cs_main);
     return
```

Naming style: PascalCase for `getChainstateForIndexing()` and `getSnapshotHeight()`?